tangelo/toolboxes/ansatz_generator/tests/test_hea.py

```diff
@@ -104,6 +106,66 @@ def test_hea_H4_doublecation(self):
         energy = sim.get_expectation_value(qubit_hamiltonian, hea_ansatz.circuit)
         self.assertAlmostEqual(energy, -0.795317, delta=1e-4)
 
+    def test_hea_circuit_variational_reference_state(self):
+        """ Verify variational gate parameters are frozen in a reference state circuit"""
+
+        # Create simple reference circuit:
+        ref_circuit = Circuit([
+            Gate('Ry', 0, parameter=1.0, is_variational=True),
+            Gate('Ry', 1, parameter=1.0, is_variational=True),
+            Gate('CX', 0, 1),
+        ])
+
+        # create HEA ansatz with reference circuit
+        hea_ansatz = HEA(n_qubits=2, reference_state=ref_circuit)
+        hea_circ = hea_ansatz.build_circuit()
+        hea_circ_gates = list(hea_circ)
+
+        # ensure gates are correctly prepended to circuit
+        self.assertEqual(hea_circ.width,2)
+        self.assertEqual(hea_circ_gates[0].name, 'RY')
```

Comment:
Could we replace the asserts with calls to `all` ? I think that would be cleaner and more generic. Maybe something like

```
ref_circ_gates = list(ref_circ)
all([hea_circ_gates[i].name == ref_circ_gates[i].name for i in hea_circ_gates.size])
all([g.is_variational == False for g in hea_circ])
```

It feels like it's using a few useful properties and magic methods from the `Circuit` class

Reply:
Fixed in [commit 37a9497](https://github.com/goodchemistryco/Tangelo/pull/398/commits/37a94975fdc86946aeb7c4a7e5a0f9a7555aa9b7). I've used `assertListEqual` so that if the gate name comparison fails, the programmer will know which specific gates are incorrectly named.